### PR TITLE
Streaming track results for final_runbook on AzureD8lds_v5

### PR DESCRIPTION
**Pull request by harsha-simhadri**

### CODE REVIEW

**Comment by nk2014yj** [comment]
https://github.com/harsha-simhadri/big-ann-benchmarks/pull/206

puck is not in this results. Could you add puck to it？

File: neurips23/streaming/res_final_runbook_AzureD8lds_v5.csv [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+puck,"Puck('C200_F200_FN8_Flat_filter_topk1900', {'radius_rate': 1.0, 'search_coarse_count': 200})",msturing-30M-clustered(final_runbook.yaml),10,0.0,0.047180891036987305,162396.0,0.0,0.0,streaming,0.9793199999999949
+puck,"Puck('C200_F200_FN8_Flat_filter_topk2000', {'radius_rate': 1.0, 'search_coarse_count': 200})",msturing-30M-clustered(final_runbook.yaml),10,0.0,0.044286489486694336,162268.0,0.0,0.0,streaming,0.9805199999999994
+puck,"Puck('C200_F200_FN8_Flat_filter_topk2100', {'radius_rate': 1.0, 'search_coarse_count': 200})",msturing-30M-clustered(final_runbook.yaml),10,0.0,0.04399824142456055,162216.0,0.0,0.0,streaming,0.9829200000000002
+puck,"Puck('C200_F200_FN8_Flat_filter_topk2200', {'radius_rate': 1.0, 'search_coarse_count': 200})",msturing-30M-clustered(final_runbook.yaml),10,0.0,0.04267311096191406,162304.0,0.0,0.0,streaming,0.9842099999999988
+puck,"Puck('C200_F200_FN8_Flat_filter_topk2300', {'radius_rate': 1.0, 'search_coarse_count': 200})",msturing-30M-clustered(final_runbook.yaml),10,0.0,0.0505528450012207,162288.0,0.0,0.0,streaming,0.9855499999999922
+
+* puck was ruin with 16GB docker memory limit, rather than 8GB.
```
Comment (ingberam) [inline neurips23/streaming/res_final_runbook_AzureD8lds_v5.csv:17]: run (not ruin)

**Review by ingberam — COMMENTED** [review]
can you also log the running times? so that we know how the winner is chosen?(mostly relevant for puck)

**Comment by nk2014yj** [comment]
I would like to confirm whether the results of puck are valid on this track, as I have seen a note indicating that puck was executed on a 16GB memory machine.

I added the following restrictions on my 16G EC2 instance of AWS, and puck was able to execute smoothly. Thank you
image

**Comment by Mysshuaige** [comment]
Hi @harsha-simhadri, we found this version of Puck's config.yaml in PR#206 which seems to have passed the competition deadline, and this set of parameters seems to have been updated in November.
微信图片_20231122114932

**Comment by nk2014yj** [comment]
> Hi @harsha-simhadri, we found this version of Puck's config.yaml in PR#206 which seems to have passed the competition deadline, and this set of parameters seems to have been updated in November. 微信图片_20231122114932

I checked the submission time of other algorithms, and it may be due to time differences, many of them were submitted on November 1st.

On November 1st, the search parameter version of puck is this link：https://github.com/harsha-simhadri/big-ann-benchmarks/pull/206/commits/f9337dbaf4792826905730db7e3cd6dca8bf833b

For this version, the score of puck is 0.9793199999999949. I accept the parameters of the November 1st version as the final score.

**Comment by Mysshuaige** [comment]
Hi @maumueller, we've checked the time you merged the pull request. The timestamp for commit 095a08 appears to be already November 1st worldwide. What are your thoughts on this?
PRtime

**Comment by DorritTang** [comment]
https://github.com/harsha-simhadri/big-ann-benchmarks/issues/251

Please also pay attention to this issue.

**Comment by maumueller** [comment]
> Hi @maumueller, we've checked the time you merged the pull request. The timestamp for commit 095a08 appears to be already November 1st worldwide. What are your thoughts on this? PRtime

The time of the final commit by the time is relevant, not the merge date.

**Comment by Mysshuaige** [comment]
> > Hi @maumueller, we've checked the time you merged the pull request. The timestamp for commit 095a08 appears to be already November 1st worldwide. What are your thoughts on this? PRtime
> 
> @Mysshuaige The time of the final commit by the time is relevant, not the merge date.


Sorry for any confusion. What I mean to express is whether the commits made after this commit(095a08) have surpassed the competition deadline?

**Comment by 3518905** [comment]
> > > Hi @maumueller, we've checked the time you merged the pull request. The timestamp for commit 095a08 appears to be already November 1st worldwide. What are your thoughts on this? PRtime
> > 
> > 
> > @Mysshuaige The time of the final commit by the time is relevant, not the merge date.
> 
> Sorry for any confusion. What I mean to express is whether the commits made after this commit(095a08) have surpassed the competition deadline?

Most submitted algorithms have modified the training and retrieval parameters based on DiskAnn. On my own machine, several algorithms have been running for over an hour. 
As a student who focuses on this field, I think the focus should be on where the future of ann, rather than on the deadline. If any students are willing to submit it in the future, I hope it can be continuously updated.

**Comment by nathan-oceanaa** [comment]
I've been quite curious lately and have dedicated the past few days to evaluating various ANNS algorithms, aiming to understand each one. In the process, I couldn't help but notice ongoing discussions about potential rule violations among participants. 
My attention was drawn to #251 , and upon a detailed comparison of the binary packages, I noted disparities from diskann. This has sparked doubts regarding its adherence to open-source principles in the context of the code review for submission #221. Actually, I notice that submissions from team hwtl_sdu do not public their codes, including #232 . Additionally, I see that the pyanns team from Zilliz has submitted solution to streaming  #229 , and other tracks #224 , #225, #228.  A committer in their [repository](https://github.com/veaaaab/DiskANN/commits/bigann23_streaming) appears to be from @zilliztech. Given their role as organizers, it prompts questions about the fairness of the competition.
Taking into account the discussions among participants, these are just some thoughts for participants and the committee.

**Comment by harsha-simhadri** [comment]
> I would like to confirm whether the results of puck are valid on this track, as I have seen a note indicating that puck was executed on a 16GB memory machine.
> 
> I added the following restrictions on my 16G EC2 instance of AWS, and puck was able to execute smoothly. Thank you image

@nk2014yj Yes, it does run with 8GB mem_limit and memswap_limit on docker, but only on a machine with 16GB DRAM. On a machine with 8GB DRAM, it does fail. There is a bug in the framework affecting mem_limit passed to docker, and I patched that locally and ensured the intended memory limits are passed to docker. However, I don't think docker is actually restricting memory used effectively -- your submission is over the limit but docker runs anyway. Either way, I will post the results since it is our fault that we did not test whether docker limited memory correctly.

**Comment by harsha-simhadri** [comment]
> > > > Hi @maumueller, we've checked the time you merged the pull request. The timestamp for commit 095a08 appears to be already November 1st worldwide. What are your thoughts on this? PRtime
> > > 
> > > 
> > > @Mysshuaige The time of the final commit by the time is relevant, not the merge date.
> > 
> > 
> > Sorry for any confusion. What I mean to express is whether the commits made after this commit(095a08) have surpassed the competition deadline?
> 
> Most submitted algorithms have modified the training and retrieval parameters based on DiskAnn. On my own machine, several algorithms have been running for over an hour. As a student who focuses on this field, I think the focus should be on where the future of ann, rather than on the deadline. If any students are willing to submit it in the future, I hope it can be continuously updated.

I would welcome ongoing submissions after NeurIPS'23. We welcome volunteers who could help with this.

**Comment by harsha-simhadri** [comment]
> I've been quite curious lately and have dedicated the past few days to evaluating various ANNS algorithms, aiming to understand each one. In the process, I couldn't help but notice ongoing discussions about potential rule violations among participants. My attention was drawn to #251 , and upon a detailed comparison of the binary packages, I noted disparities from diskann. This has sparked doubts regarding its adherence to open-source principles in the context of the code review for submission #221. Actually, I notice that all submissions from team hwtl_sdu do not public their codes, including #232 . Additionally, I see that the pyanns team from Zilliz has submitted solution to streaming #229 , and other tracks #224 , #225, #228. A committer in their [repository](https://github.com/veaaaab/DiskANN/commits/bigann23_streaming) appears to be from @zilliztech. Given their role as organizers, it prompts questions about the rules. Taking into account the discussions among participants, these are just some thoughts for participants and the committee.

I intend for this to be an open framework, and encourage people to be open about things, but cannot enforce it.
On #251, I have requested the author for clarification. Let us follow the discussion there.
On pyanns, I have had a tough time pinning down the author @veaaaab  who has not been responsive. After looking around github comments and guessing, I was able to reach @hhy3  who said @veaaaab  was their github handle. I definitely agree that the author could have been more forthcoming with their identity. We are in touch with @hhy3  to ascertain authors and affiliations of pyanns.

**Pull request merged**